### PR TITLE
gh-115765: Use AX_CHECK_DEFINE instead of AC_EGREP_*

### PR DESCRIPTION
**Pull request by erlend-aasland**
Alternative to #115791 

* Issue: gh-115765

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @mhsmith

File: configure.ac [hunk 1]
```diff
@@ -4517,34 +4509,21 @@ if test "$ipv6" = yes -a "$cross_compiling" = no; then
...
+                            [ipv6type=$i
+                             ipv6lib=inet6
+                             ipv6libdir=/usr/local/v6/lib
+                             ipv6trylibc=yes])
 			;;
 		linux-glibc)
-			dnl http://www.v6.linux.or.jp/
-			AC_EGREP_CPP([yes], [
-#include <features.h>
-#if defined(__GLIBC__) && ((__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1) || (__GLIBC__ > 2))
```
Comment (erlend-aasland) [inline configure.ac:4543]: Glibc 2.1 was released in February 1999. According to DistroWatch, we can safely assume that we run on glibc 2.1 or newer. All we need here is to check for `__GLIBC__`, AFAICS.

**Review by mhsmith — APPROVED** [review]

**Review by corona10 — APPROVED** [review]
lgtm!

**Comment by erlend-aasland** [comment]
Thanks for the review, both of you! Highly appreciated :)

**Pull request merged**